app/component/search/search-two-fields-container.cjsx

```diff
@@ -120,22 +147,79 @@ class SearchTwoFieldsContainer extends React.Component
         endpoint={destination}
         geolocation={geolocation}
         onClick={(e) =>
-          e.preventDefault()
-          @context.executeAction SearchActions.openDestinationSearch,
-            position: destination
-            placeholder: destinationPlaceholder
-          focusInput()
-        }
+          @setState
+            selectedTab: "destination"
+            modalIsOpen: true
+          @context.executeAction SearchActions.executeSearch, @context.getStore('EndpointStore').getDestination()?.address || ""
+          @focusInput("destination")}
         autosuggestPlaceholder={destinationPlaceholder}
-        navigateOrInputPlaceHolder={@context.intl.formatMessage(
-          id: 'give-destination'
-          defaultMessage: 'Type destination')}
         id='destination'
       />
 
     <div>
       <SearchTwoFields from={from} to={to} onSwitch={@onSwitch} routeIfPossible={@routeIfPossible}/>
-      <SearchModal ref="modal" initialPosition={destination}/>
+      <SearchModal
+        ref="modal"
+        selectedTab={@state.selectedTab}
+        modalIsOpen={@state.modalIsOpen}
+        closeModal={@closeModal}>
+        <Tab
+          className="search-header__button"
+          label={originSearchTabLabel}
+          ref="searchTab"
+          value={"origin"}
+          onActive={@onTabChange}
+          style={{
+            color: if @state.selectedTab == "origin" then "#333" else "#7f929c",
+            fontSize: "11px",
+            fontFamily: "Gotham Rounded SSm A, Gotham Rounded SSm B, Arial, Georgia, Serif",
```

Comment:
Yes, they should. But I didn't find a way to do it without !important. And as we discussed about this earlier, we decided we would then use inline styles. Though now I discussed it with Samuli, and we agreed that we could use !important this time.